test/boost/view_schema_test.cc

```diff
@@ -3011,6 +3011,21 @@ struct update_counter {
 };
 
 SEASTAR_TEST_CASE(test_view_update_generating_writetime) {
+    // In this test, we verify that generating view updates abides by the intended algorithm.
+    // We test two materialized views whose semantics may differ:
+    //
+    // * mv1: its primary key is a permutation of the base table's primary key. Because of that,
+    //        it will have virtual columns corresponding to unselected columns from the base table.
+    //        Creating a value in such a column (in the base table) will generate a view update
+    //        to the MV. However, updating it will not generate an update UNLESS it changes
+    //        the cell's TTL.
+    // * mv2: its primary key consists of the columns from the base table's primary key and one
+    //        regular column. Because of that, the MV will NOT have any virtual columns corresponding
+    //        to the unselected columns from the base table. As a result, no view updates will be
+    //        generated for unselected columns as a result.
+    //
+    // We also verify that `WRITETIME` works correctly.
```

Comment:
This sentence is the main goal of this test and I think it should come first. However, it's not about testing the `WRITETIME` function like your phrasing might suggest to readers. Rather - it's about the timestamps written to the view table being set to the correct value (same as in the base table). But, we check **only** the timestamps that are actually readable (using `WRITETIME()`) in the view - in other words, only the timestamps of selected columns. The timestamp of columns not readable with `WRITETIME()` - namely virtual columns (created in some cases for unselected columns) do not matter, so Scylla can - and does - avoid updating them, and may skip some view updates altogether.  The second goal of this test is to verify that view updates are indeed skipped when their only function would have been to update a non-readable timestamp.

Reply:
I think I get it now, thank you for your explanation!